Remove serde-based serialization

bobbinth opened the pull request:
This PR removes `serde`-based serialization (and `serde` dependency) from `miden-objects` structs. With the latest refactorings of the testing structure, we no longer need these for tests, and I don't think they are used anywhere else (but we should confirm).

Fumuran reviewed (approved):
Looks good to me!

Probably `bench-tx` will be the only crate which will use `serde`, but it doesn't serialize any structure from `objects`, so it should work fine.

SantiagoPittella reviewed (approved):
We have been testing it in the client side and is working properly, though we rebased with next to make it work.

bobbinth commented:
> We have been testing it in the client side and is working properly, though we rebased with next to make it work.

Great! Thank you! I've rebased - so, this branch should be up to date with `next` now.

igamigo reviewed (approved):
LGTM!
